### PR TITLE
feat(deeplink): added more cases and redirects for deeplinks

### PR DESCRIPTION
**Pull request by pedroapfilho**
**buy (already customer)** -> `/#/open/simple-buy`

**buy (new customer)** -> `/#/open/simple-buy?crypto={BTC | ETH | ...}&amount={1 | 99 | 200 | ...}&email={test@blockchain.com | ...}&fiatCurrency={USD | GBP | ...} (if any of these fields is not filled, it will just go to /login)`

**kyc** -> `/#/open/kyc?tier={0 | 1 | 2 | ...} (the tier here is not necessary, the default is 2, that is gold member)`

**swap** -> `/#/open/swap (this doesn't cover when the user is not gold or basically can't swap)`

**interest** -> `/#/open/interest`

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.ts [hunk 1]
```diff
@@ -85,9 +85,20 @@ export default ({ api, coreSagas, networks }) => {
   }
 
   const defineKycGoal = function * (search) {
+    // /#/open/kyc?tier={1, 2, ...}
     const params = new URLSearchParams(search)
-    yield put(actions.goals.saveGoal('kyc', { tier: params.get('tier') }))
-    yield put(actions.router.push('/login'))
+
+    const tier = params.get('tier') || 2
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.ts:91]: we have tier types constants they are siting here
```
import { TIER_TYPES } from 'blockchain-wallet-v4-frontend/src/modals/Settings/TradingLimits/model'
```

we can move them to some mor generic place but would help not to use magic numbers as it is 2 ;)

File: packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.ts [hunk 2]
```diff
@@ -180,21 +193,44 @@ export default ({ api, coreSagas, networks }) => {
...
+    }
+
+    if (startsWith(DeepLinkGoal.KYC, pathname)) {
+      return yield call(defineKycGoal, search)
+    }
+
+    // TODO check why it uses includes
     // crypto send / bitpay links
-    if (includes('bitcoin', pathname))
+    if (includes(DeepLinkGoal.BITCOIN, pathname)) {
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.ts:214]: 👏

**Review by milan-bc — APPROVED** [review]
LGTM, I just added one minor comment, you may address it now or in some other PR

**Review by tbuchann — APPROVED** [review]

**Comment by pedroapfilho** [comment]
@tbuchann already checked that the last deeplinks are working. I'll push this one.

**Pull request merged**